docs/test-guidelines.md

```diff
@@ -0,0 +1,57 @@
+# Test guidelines
```

Comment:
About the file location, file name and content organization, we currently have these 2 files:
- `docs/tests/jest.md`
- `docs/testing.md`

I think if we add a third file called `docs/test-guidelines.md` might be a bit confusing over what we would expect here. Could we maybe rethink the file name or the content organization itself?

If we decide to have this information on a separate file, as in this PR, maybe we could link it on the `testing.md` file, as well as re-name it, so it's a bit more explicit what this is about?


![Screenshot from 2024-07-15 09-25-58](https://github.com/user-attachments/assets/2d0c6829-7e1b-44d7-aa5d-5f1fb91fac7d)

Reply:
humm not sure I follow. This is on the contributors doc repo.. and the screenshot you posted is from extension repo. Are you suggesting adding a link to this from the Extension repo docs? Or adding this doc to the Extension repo docs instead of adding it here?

Reply:
I like how this document broadly covers the purpose of testing and how we use it at MetaMask.

Are you familiar with the [Diataxis](https://diataxis.fr/) system? This is a way of arranging information that I found while during research prior to setting up this repo. Basically it suggests that there are different kinds of documentation that serve different purposes. When I wrote the unit testing guidelines I provided some background information for each guideline but I tried to keep the overall document squarely in the "reference" category.

This document seems more of an "explanation" kind of document. That's great, I think we've reached the point where we need some things like this. I just wanted to point out that perhaps it's worth thinking about that distinction, and I wonder if we ought to gear this document toward that. I notice that the name of this file is "Test _guidelines_" and how this file lists some guidelines for each testing category but also provides a summary. I wonder if we want to keep all guidelines in separate documents and just use this file to provide a general overview for each kind of testing category?

In addition, perhaps we have a `testing/` directory and we rename this to `testing/overview.md` or something like that, and then we can relocate the existing `unit-testing.md` to `testing/unit-testing-guidelines.md` or something?

There are many ways to skin this cat, so to speak, so I'm curious to know what you think.

Reply:
Thanks for this @mcmire. I was not familiar with the Diataxis system. It seems to be a great guide to differentiate types of documentation. I looked into the "[explanation](https://diataxis.fr/explanation/)" details and tried to update the current `overview` to be more inline with it.
Also, as you mentioned, I took the opportunity to re-organise the test documentation. Please have a look and let me know your thoughts.